pkg/query/querier.go

```diff
@@ -54,28 +60,26 @@ func NewQueryableCreator(logger log.Logger, reg prometheus.Registerer, proxy sto
 
 type queryable struct {
 	logger               log.Logger
-	reg                  prometheus.Registerer
 	replicaLabels        []string
 	storeMatchers        [][]storepb.LabelMatcher
 	proxy                storepb.StoreServer
 	deduplicate          bool
 	maxResolutionMillis  int64
 	partialResponse      bool
 	skipChunks           bool
-	gateKeeper           *gate.Keeper
+	gateFn               func() gate.Gate
```

Comment:
non-blocker suggestion: How about `gateProvider` or `gateProviderFn`?
```suggestion
	gateProvider               func() gate.Gate
```